src/weight_windows.cpp

```diff
@@ -0,0 +1,284 @@
+#include "openmc/error.h"
+#include "openmc/file_utils.h"
+#include "openmc/particle.h"
+#include "openmc/particle_data.h"
+#include "openmc/search.h"
+#include "openmc/weight_windows.h"
+#include "openmc/xml_interface.h"
+
+namespace openmc {
+
+// read the weight window file
+void read_weight_window_xml() {
+  using namespace weight_window;
+  using namespace settings;
+  using namespace pugi;
+
+  // Check if weight_window.xml exists                                                         
+  std::string filename = path_input + "weight_windows.xml";
+  
+  // Parse settings.xml file                                                              
+  xml_document doc;                                                                       
+  if (file_exists(filename)) {
+    auto result = doc.load_file(filename.c_str());                                          
+    if (!result) {                                                                          
+      fatal_error("Error processing weight_windows.xml file.");                                   
+    }
+    // Get root element                                                                     
+    xml_node root = doc.document_element();     
+    std::cout << " Reading weight window XML file ..." << std::endl;
+    // try and read a weight window
+    settings::ww_settings = std::make_shared<weight_window::WeightWindow>(root);
+  } else {
+    settings::ww_settings = std::make_shared<weight_window::WeightWindow>();
+  }                                                                       
+}
+
+//==============================================================================
+// WeightWindowMesh implementation
+//==============================================================================
+
+using namespace weight_window;
+
+WeightWindow::WeightWindow() 
+{
+  // default parameters
+  n_ww = false;                  // flag for neutron use weight window
+  p_ww = false;                  // flag for photon use weight window
+  user_defined_biasing = false;  // flag for weight biasing in energy
+
+  // upper weight window = upper_ratio * lower weight window
+  weight_params[ParticleType::neutron].upper_ratio = 5.; 
+  // survival weight = survival_ratio * lower weight window  
+  weight_params[ParticleType::neutron].survival_ratio = 3.; 
```

Comment:
Surely there is some way to remove the redundant code here. My recommendation is to have the default constructor of whatever's stored in `weight_params` use these reasonable defaults. While it is a `struct` and not a `class`, it would remove the need for all this code that's repeated (especially if we ever start moving other particles).